onnxruntime/python/tools/quantization/matmul_weight4_quantizer.py

```diff
@@ -0,0 +1,138 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+from typing import List, Tuple
+
+import numpy as np
+import onnx
+from onnx.onnx_pb import GraphProto, ModelProto, NodeProto, TensorProto
+
+from .onnx_model import ONNXModel
+from .q4dq_wrapper import Q4dqWrapper
+from .quant_utils import attribute_to_kwarg
+
+
+class MatMulWeight4Quantizer:
+    """Perform 4b quantization of constant MatMul weights"""
+
+    ##################
+    # quantization types, must be consistent with native code type
+    # MLAS_BLK_QUANT_TYPE defined in mlas_q4.h
+
+    # 32 number block, symmetric quantization, with one fp32 as scale, zero point is always 0
+    BlkQ4Sym = 0
+
+    # 32 number block, quantization, with one fp32 as scale, one uint8 zero point
+    BlkQ4Zp8 = 1
+
+    def __init__(self, model: ModelProto, q4dq: Q4dqWrapper, quant_type: int):
```

Comment:
>Q4dqWrapper [](http://example.com/codeflow?start=48&length=11)

As we discussed, there will be a following PR to implement Q4dqWrapper with python code in quantization tool.

Reply:
you want to have a comments on the code about possible future PR?